jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/collector/VaultInventoryCollector.java

```diff
@@ -65,18 +70,39 @@ public List<VaultVH> fetchVaultDetails(SubscriptionVH subscription) throws Excep
 						vaultVH.setSku((Map<String, Object>) propertiesMap.get("sku"));
 						vaultVH.setVaultUri(propertiesMap.get("vaultUri").toString());
 						JsonArray accessPolicies = properties.getAsJsonArray("accessPolicies");
-						JsonObject permissions = (JsonObject) ((JsonObject) accessPolicies.get(0)).get("permissions");
-						HashMap<String, List<String>> permissionsMap = new Gson().fromJson(permissions.toString(),
-								HashMap.class);
-						vaultVH.setPermissionForKeys(permissionsMap.get("keys"));
-						vaultVH.setPermissionForSecrets(permissionsMap.get("secrets"));
-						vaultVH.setPermissionForCertificates(permissionsMap.get("certificates"));
+						if(accessPolicies.size()>0) {
+							JsonObject permissions = (JsonObject) ((JsonObject) accessPolicies.get(0)).get("permissions");
+							HashMap<String, List<String>> permissionsMap = new Gson().fromJson(permissions.toString(),
+									HashMap.class);
+							vaultVH.setPermissionForKeys(permissionsMap.get("keys"));
+							vaultVH.setPermissionForSecrets(permissionsMap.get("secrets"));
+							vaultVH.setPermissionForCertificates(permissionsMap.get("certificates"));
+						}
 					}
 					if (tags != null) {
 						HashMap<String, Object> tagsMap = new Gson().fromJson(tags.toString(), HashMap.class);
 						vaultVH.setTags(tagsMap);
 					}
+					String id =vaultVH.getId();
+					int beginningIndex=id.indexOf("resourceGroups")+15;
+					String resourceGroupName=(vaultVH.getId()).substring(beginningIndex,id.indexOf('/',beginningIndex+2));
+					log.debug("Resource group name: {}",resourceGroupName);
+					vaultVH.setResourceGroupName(resourceGroupName);
+					try
+					{
+						Vault azureVault=azure.vaults().getById(id);
+					//				.keys().list().get(0).attributes().expires()
 
+						PagedList<Key> keys=azureVault.keys().list();
+						for(Key key:keys)
+						{
+							vaultVH.setKeyExpirationDate(key.attributes().expires().toString());
```

Comment:
We should collect the data for all keys to check expiration date for all the keys.

Reply:
made changes